### DISCUSSION TITLE
help 😥

### DISCUSSION THREAD

**Discussion by Naiml007** [Ani-Moopa/Moopa#84] [General]
![image](https://github.com/Ani-Moopa/Moopa/assets/90306726/fd617a3b-5d45-47cf-9857-2e197f428d88)

auth 
![image](https://github.com/Ani-Moopa/Moopa/assets/90306726/76268588-1a30-4a2c-9390-69fe75dc5075)
![image](https://github.com/Ani-Moopa/Moopa/assets/90306726/7fd10480-246b-4c3a-b8ce-f8e07079e141)
![image](https://github.com/Ani-Moopa/Moopa/assets/90306726/c67a52e1-d799-4672-9f1f-e09cac39cdb7)

**Comment by Naiml007** [comment]
https://ani-moopa-five.vercel.app/en/

**Comment by DevanAbinaya** [comment]
> https://cors.moopa.my.id 

this cors doesn't work anymore, deploy your own cors proxy [here](https://github.com/Gratenes/m3u8CloudflareWorkerProxy)

> ![image](https://github.com/Ani-Moopa/Moopa/assets/97084324/af90dc7e-23dd-4450-83a7-13f8a4beafa5)

make sure that the redirect url is canonical to your current domain url
> ![image](https://github.com/Ani-Moopa/Moopa/assets/97084324/6d3d51d3-9920-45b8-b7d9-e67dd5ab5684)

**Comment by Naiml007** [comment]
What URL do I have to put inside cors proxy?

**Reply by DevanAbinaya** [reply]
wdym, what else other then video url.

**Reply by Naiml007** [reply]
help! what i have to do here 
![image](https://github.com/Ani-Moopa/Moopa/assets/90306726/5c615d91-f5af-41ea-a4a0-ddae6fb806db)

**Reply by Naiml007** [reply]
@DevanAbinaya  brooo

**Reply by DevanAbinaya** [reply]
You don't need to modify any code inside the proxy itself, all you just gotta do is put your deployed proxy url inside your .env file, sorry I slept early last night.

**Reply by Naiml007** [reply]
it's ok 👌

**Comment by Naiml007** [comment]
Is my proxy URL right? https://m3u8proxy.nai*******.workers.dev/https://ani-moopa-five.vercel.app

**Reply by DevanAbinaya** [reply]
that's not how it works... 
do you know what **Environment Variables** is?

**Reply by Naiml007** [reply]
yea i know env where you can store your secret tokens

**Reply by Naiml007** [reply]
I just asked you if that was the right format.

**Reply by Naiml007** [reply]
also im new at hls staff

**Reply by DevanAbinaya** [reply]
here 
![image](https://github.com/Ani-Moopa/Moopa/assets/97084324/46761547-ae7e-40ba-b0ba-044ab5dc5f1e)

**Reply by DevanAbinaya** [reply]
### you put it like this 
![image](https://github.com/Ani-Moopa/Moopa/assets/97084324/51b79136-2c00-4661-872e-b83a32832de6)

**Reply by Naiml007** [reply]
Tnx ♥️

**Reply by Naiml007** [reply]
still not working 
![image](https://github.com/Ani-Moopa/Moopa/assets/90306726/06538f5b-e23a-4fd1-92d5-2af07044a0b7)

**Comment by Naiml007** [comment]
https://ani-moopa-five.vercel.app/

**Comment by Naiml007** [comment]
![image](https://github.com/Ani-Moopa/Moopa/assets/90306726/c44df367-2596-43af-92f0-e6febeaef4d5)

**Comment by Naiml007** [comment]
Request URL:
https://m3u8proxy.naimurdj9.workers.dev/proxy/m3u8/https%3A%2F%2Fwww045.vipanicdn.net%2Fstreamhls%2F784821fabdeaff7592e0c178594b1b94%2Fep.1.1696432433.m3u8/%7B%22Referer%22%3A%22https%3A%2F%2Fgoone.pro%2Fembedplus%3Fid%3DMjEyODg0%26token%3DxWrCIm9vkPWX_8vsHxSB-w%26expires%3D1697056349%22%7D



Invalid URL

**Comment by Naiml007** [comment]
proxy url responding m3u8 video urls

![image](https://github.com/Ani-Moopa/Moopa/assets/90306726/8be2ee73-c5d4-4310-9895-499b6f0f3459)

**Reply by DevanAbinaya** [reply]
update this [code](https://github.com/Ani-Moopa/Moopa/blob/77eb41dcf3c63a7e3539d1fc6fcf2ca2f5e116a4/components/watch/player/playerComponent.js#L94-L109) first with the following code :
```js
const referer = data?.headers?.Referer;
        const source = data.sources.map((items) => {
          const isDefault =
            provider !== "gogoanime"
              ? items.quality === "default" || items.quality === "auto"
              : resolution === "auto"
              ? items.quality === "default" || items.quality === "auto"
              : items.quality === resolution;
          return {
            ...(isDefault && { default: true }),
            html: items.quality === "default" ? "adaptive" : items.quality,
            url:
              provider === "gogoanime"
                ? `https://cors.moopa.workers.dev/?url=${encodeURIComponent(
                    items.url
                  )}${referer ? `&referer=${encodeURIComponent(referer)}` : ""}`
                : `${proxy}${items.url}`,
          };
});
 ```

**Reply by Naiml007** [reply]
tnx broo

**Reply by Naiml007** [reply]
!!!!!!!!!!!!!!!!!!
![image](https://github.com/Ani-Moopa/Moopa/assets/90306726/522e4d8e-8e1d-4bf4-8271-baa804641402)

**Reply by Naiml007** [reply]
![image](https://github.com/Ani-Moopa/Moopa/assets/90306726/95dfd08c-b8c5-45d8-a19c-5386b69e0ea0)

**Reply by DevanAbinaya** [reply]
Change the cors to your proxy
```https://cors.moopa.workers.dev/ >> https://your.cors.tld/```

**Reply by DevanAbinaya** [reply]
> update this [code](https://github.com/Ani-Moopa/Moopa/blob/77eb41dcf3c63a7e3539d1fc6fcf2ca2f5e116a4/components/watch/player/playerComponent.js#L94-L109) first with the following code :
> ```js
> const referer = data?.headers?.Referer;
>         const source = data.sources.map((items) => {
>           const isDefault =
>             provider !== "gogoanime"
>               ? items.quality === "default" || items.quality === "auto"
>               : resolution === "auto"
>               ? items.quality === "default" || items.quality === "auto"
>               : items.quality === resolution;
>           return {
>             ...(isDefault && { default: true }),
>             html: items.quality === "default" ? "adaptive" : items.quality,
>             url:
>               provider === "gogoanime"
>                 ? `https://cors.moopa.workers.dev/?url=${encodeURIComponent(
>                     items.url
>                   )}${referer ? `&referer=${encodeURIComponent(referer)}` : ""}`
>                 : `${proxy}${items.url}`,
>           };
> });
>  ```

In this code not on vercel

**Reply by Naiml007** [reply]
tnx brooo 😊

**Reply by Naiml007** [reply]
Bro, it's working now. TNX

😊
